tapa/graphir_conversion/add_iface.py

```diff
@@ -323,6 +327,17 @@ def get_graphir_iface(  # noqa: PLR0914
         ),
     ]
 
+    # set role on all interfaces
+    modules = project.modules.module_definitions
+    for module in modules:
+        if module.name not in ifaces:
+            continue
+
+        updated_ifaces = [
+            set_iface_role(module, iface) for iface in ifaces[module.name]
+        ]
+        ifaces[module.name] = list(updated_ifaces)
```

Comment:
There is no need making a copy of list?

```suggestion
        ifaces[module.name] = [
            set_iface_role(module, iface) for iface in ifaces[module.name]
        ]
```